R/commonTTest.R

```diff
@@ -443,3 +444,108 @@ summarySEwithin <- function(data=NULL, measurevar, betweenvars=NULL, withinvars=
   # Combine the un-normed means with the normed results
   merge(datac, ndatac)
 }
+
+.descriptivesPlotsRainCloudFill <- function(dataset, variable, groups = NULL, yLabel, dependency = c("none", "paired", "repeated"), horiz) {
+  # Adapted under the MIT license from:
+  # van Langen, J. (2020). Open-visualizations in R and Python.
+  # https://github.com/jorvlan/open-visualizations
+  #
+  # Permission is hereby granted, free of charge, to any person obtaining a copy
+  # of this software and associated documentation files (the "Software"), to deal
+  # in the Software without restriction, including without limitation the rights
+  # to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  # copies of the Software, and to permit persons to whom the Software is
+  # furnished to do so, subject to the following conditions:
+  #
+  # The above copyright notice and this permission notice shall be included in all
+  # copies or substantial portions of the Software.
+
+  n <- nrow(dataset)
+  if (dependency == "paired") {
+    y   <- unlist(dataset)
+    grp <- rep(as.vector(variable), c(n,n))
```

Comment:
As far as I am aware of, there is no inidicator variable to which measurement each data point belongs, so I had to create that by hand. The independent t-tests and Anovas get that from the grouping variable.

Reply:
Ok, my question was specifically about `rep(... c(n, n))` - as far I am aware `times` argument cannot be a vector??

Reply:
I think it can be actually, at least so says the help file of the function: 

> an integer-valued vector giving the (non-negative) number of times to repeat each element if of length length(x), or to repeat the whole vector if of length 1. Negative or NA values are an error. A double vector is accepted, other inputs being coerced to an integer or double vector.

Reply:
Ah, my bad, I thought that `variable` is the data, but it is the names of the two measurements... ok. Alternatively, this can be also done as `rep(variable, each=n)` (in case you have multiple repeated measurements stored as a wide format).